build-utils/mount-partitions.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/sh
+
+# Mounts boot, state, & data partitions from balenaOS.
+# The container must be privileged for this to function correctly.
+
+use_legacy_message="Using legacy host mounts."
+
+# Get the current boot device in case there are duplicate partition labels
+# for `(balena|resin)-(boot|state|data)` found. Because Supervisor is started with
+# `--net=host`, /etc/resolv.conf is bind-mounted by the Engine into the container,
+# so the partition where resolv.conf is located is assumed to be one of the
+# partitions of the current active device. The partition identifer `pN` is removed 
+# from the end so we can compare with the base device.
+current_boot_device=$(grep /etc/resolv.conf /proc/mounts | awk '{print substr($1,1,length($1)-2)}')
```

Comment:
This seems like too much of a hack for my taste. Maybe someone on the OS (cc @alexgg) has some ideas on how to do this more cleanly?

Reply:
@pipex in the OS we do something like:
```
findmnt --noheadings --canonicalize --uniq --output SOURCE /mnt/sysroot/active
```
Which will output the device the active partition is mounted from. Is that what you are looking for?

Reply:
Yup, that's what I was looking for, thank you @alexgg!

Reply:
This is a good idea, however doesn't work in the Supervisor container as the container only has access to the following mounts, as listed in `/proc/mounts`:
```
/usr/src/app # cat /proc/mounts | grep -v 'overlay' | grep -v 'docker' | grep -v '/mnt/root'
proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0
tmpfs /dev tmpfs rw,nosuid,size=65536k,mode=755 0 0
devpts /dev/pts devpts rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=666 0 0
sysfs /sys sysfs rw,nosuid,nodev,noexec,relatime 0 0
tmpfs /sys/fs/cgroup tmpfs rw,nosuid,nodev,noexec,relatime,mode=755 0 0
cgroup /sys/fs/cgroup/systemd cgroup rw,nosuid,nodev,noexec,relatime,xattr,name=systemd 0 0
cgroup /sys/fs/cgroup/blkio cgroup rw,nosuid,nodev,noexec,relatime,blkio 0 0
cgroup /sys/fs/cgroup/cpu,cpuacct cgroup rw,nosuid,nodev,noexec,relatime,cpu,cpuacct 0 0
cgroup /sys/fs/cgroup/devices cgroup rw,nosuid,nodev,noexec,relatime,devices 0 0
cgroup /sys/fs/cgroup/pids cgroup rw,nosuid,nodev,noexec,relatime,pids 0 0
cgroup /sys/fs/cgroup/freezer cgroup rw,nosuid,nodev,noexec,relatime,freezer 0 0
cgroup /sys/fs/cgroup/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset 0 0
cgroup /sys/fs/cgroup/net_cls,net_prio cgroup rw,nosuid,nodev,noexec,relatime,net_cls,net_prio 0 0
cgroup /sys/fs/cgroup/perf_event cgroup rw,nosuid,nodev,noexec,relatime,perf_event 0 0
cgroup /sys/fs/cgroup/memory cgroup rw,nosuid,nodev,noexec,relatime,memory 0 0
mqueue /dev/mqueue mqueue rw,nosuid,nodev,noexec,relatime 0 0
shm /dev/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
/dev/nvme0n1p6 /data ext4 rw,relatime 0 0
tmpfs /var/log tmpfs rw,relatime 0 0
/dev/nvme0n1p1 /boot/config.json vfat rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro 0 0
/dev/nvme0n1p6 /boot/apps.json ext4 rw,relatime 0 0
proc /mnt/fib_trie proc rw,relatime 0 0
/dev/nvme0n1p6 /etc/resolv.conf ext4 rw,relatime 0 0
/dev/nvme0n1p6 /etc/hostname ext4 rw,relatime 0 0
/dev/nvme0n1p6 /etc/hosts ext4 rw,relatime 0 0
/dev/disk/by-state/resin-state /usr/share/ca-certificates/balena ext4 rw,relatime 0 0
tmpfs /run/balena-engine.sock tmpfs rw,nosuid,nodev,mode=755 0 0
```

This is excluding `/mnt/root` which we don't want to rely on, as well as any overlay mounts resulting from docker or otherwise. Furthermore, we need to exclude `/boot` and `/data` as those are also mounts provided by the OS and shouldn't be relied on. Based on this, `/etc/resolv.conf`, `/etc/hostname`, and `/etc/hosts` look to be the only ones we can rely on to look for the current boot device.

Reply:
`fdisk -l` [might be](https://askubuntu.com/a/952789) an alternative? 

```
/usr/src/app # fdisk -l | grep '*  ' | cut -d' ' -f1 2>&1
/dev/mmcblk0p1
```

Reply:
Good shout, that works! Thanks 👍🏼

Reply:
Switched to `fdisk` command in [latest force-push](https://github.com/balena-os/balena-supervisor/pull/2120/commits/71cddb6a1b9d51a59a350726a80035e5afce1403#diff-8a96bc9af661488f86b92785e9382315f61c548974c4bcce92c1a1deb9a129f8R11)